app/main/views/briefs.py

```diff
@@ -174,19 +174,22 @@ def update_brief_status(brief_id, action):
     brief = Brief.query.filter(
         Brief.id == brief_id
     ).first_or_404()
-
-    if brief.framework.status != 'live':
-        abort(400, "Framework is not live")
+    framework_status = brief.framework.status
 
     action_to_status = {
         'publish': 'live',
         'withdraw': 'withdrawn',
         'cancel': 'cancelled',
         'unsuccessful': 'unsuccessful'
     }
-    if brief.status != action_to_status[action]:
-        previousStatus = brief.status
-        brief.status = action_to_status[action]
+    new_status = action_to_status[action]
+
+    if new_status in ['live', 'withdrawn'] and framework_status != 'live':
+        abort(400, "Framework is not live")
```

Comment:
Would be good to have a test around this case as it's a new bit of logic. Previously if the framework was not live, you couldn't use this endpoint at all - now you can if the update is to either `cancelled` or `unsuccessful`.

Reply:
Updated existing test for this

Reply:
The point that David made is that a brief can't be 'closed' or 'expired' unless its corresponding framework is 'open' or 'expired'. This means that all the validation we need is done in the status setter on the brief model when it checks that we cannot move to 'cancelled' or 'unsuccessful' for any status other than 'closed'.

Reply:
@samuelhwilliams added a test to ensure we change when on the expected framework statuses